src/components/Profile.tsx

```diff
@@ -21,17 +21,18 @@ import {
   WalletIcon,
 } from './icons';
 import ConnectModal from './modals/ConnectModal';
+import useLogo from '@src/hooks/useLogo';
 
 // eslint-disable-next-line react/display-name
 const CustomPopoverButton = forwardRef<
   HTMLButtonElement,
-  { children?: ReactElement }
+  { children?: ReactElement; logo?: string }
 >((props, ref) => {
   return (
     <Button
       forwardRef={ref}
       buttonType={ButtonType.PRIMARY}
-      icon={<ConnectIcon className="size-4" />}
+      icon={props.logo ? <img src={`https://arweave.net/${props.logo}`} alt="Logo" className="size-4" /> : <ConnectIcon className="size-4" />}
```

Comment:
Is this supposed to be square?

Reply:
Makes no assumptions on logo shape. Current default icon is squarish. Can review shape in future PR.